functions/concepts/retry-timeout/src/main/java/functions/RetryTimeout.java

```diff
@@ -42,25 +41,19 @@
   @Override
   public void accept(PubSubMessage message, Context context) {
     ZonedDateTime utcNow = ZonedDateTime.now(ZoneOffset.UTC);
-    ZonedDateTime timestamp = utcNow;
+    ZonedDateTime timestamp = ZonedDateTime.parse(context.timestamp());
```

Comment:
Shouldn't there be some kind of comment discussing these two and why they are different and when you want to use one and not the other?

Reply:
@lesv I suspect @eamonnmcmanus meant to delete the `utcNow` declaration?

Reply:
Still, there are two sources of truth, and we got it wrong the first time, so that deserves an explanation, since other users are likely to make the same error unless they are told why.

Reply:
Ah - I misread, he's comparing `utcNow` to the parsed timestamp. (`utcNow` was previously the _default timestamp value_ in case one wasn't included in the Pub/Sub message.)

This LGTM. ✅

Reply:
I think the situation is pretty confusing here. I don't know under what circumstances, if any, the `messageId` and `publishTime` properties of the payload are set. I just deployed a pub/sub function and sent a message to it, and both these fields were null. We may have a much bigger cleanup to do to get this straight, though hopefully the migration to CloudEvents should make things more consistent.

Anyway, let's get this fix in for now, and try to improve things further afterwards.

Reply:
k - I'd still rather you commented about it in the source, but it's up to you.